### DISCUSSION TITLE
Option to define language chunk filenames

### DISCUSSION THREAD

**Discussion by silverwind** [microsoft/monaco-editor#3769] [Q&A] [answered]
The language chunk names like `1.js` are ugly and if their order ever changes, would present a caching issue. Suggesting a new option `languageFilename` with default value `'[name].js'` which would emit names like `javascript.js` which can safely be cached using such filenames.

**Comment by dmnsgn** [comment]
@alexdima would that be complex to implement? Reading the codebase, I am unsure where the chunk generation happens for languages. I assume from here:

https://github.com/microsoft/monaco-editor-webpack-plugin/blob/0c36fc7eb7d7dd7ed9e223e86b3d3bf892141f5f/src/loaders/include.ts#L13-L21

But since those are require call, I don't see how we can add webpackChunkName magic comments here.

**Comment by silverwind** [comment]
I think the issue may be that this plugin does not pass any filename for the chunks so webpack automatically assigns some. It seems [`output.chunkFilename`](https://webpack.js.org/configuration/output/#outputchunkfilename) can influence the chunk naming but it's rather inflexible because it [can't be specified as a function](https://github.com/webpack/webpack.js.org/issues/2070) in webpack.

**Comment by silverwind** [comment, marked as answer]
FWIW, Webpack 5 allows a function for `output.chunkFilename` so this should now be workaroundable.

**Reply by alanhe421** [reply]
any example?

**Reply by silverwind** [reply]
https://github.com/go-gitea/gitea/blob/ebb98d72462d3661caf8f5e2ca22c1ece5425409/webpack.config.js#L89-L92

**Reply by alanhe421** [reply]
solved it. thank you.

**Comment by silverwind** [comment]
Webpack 5 emits better autogenerated names, but still pretty ugly:

```
node_modules_monaco-editor_esm_vs_basic-languages_css_css_js
node_modules_monaco-editor_esm_vs_language_css_cssMode_js
```

It can be remedied with `chunkFilename` but it's rather clumsy and I'd expect this module to do it.
